### ISSUE TITLE
Potential bug in neighborhood assignments

### ISSUE THREAD

**Issue by ngreenwald** [angelolab/ark-analysis#968]
*Please refer to our [FAQ](https://ark-analysis.readthedocs.io/en/latest/_rtd/faq.html) and look at our [known issues](https://github.com/angelolab/ark-analysis/issues) before opening a bug report.*

**Describe the bug**
I'm running into some weird behavior with the neighborhood analysis script. Specifically, it seems like cells with very similar neighborhoods are being assigned to different clusters. 

For example, in the upper right hand corner, all of the blue cancer cells seem to have almost exactly the same neighbors
<img width="1472" alt="image" src="https://user-images.githubusercontent.com/13770365/228931654-c8fc5247-c69c-4210-a28a-0c1d15216cf3.png">

However, they are assigned to different neighborhoods in the output. 
<img width="1012" alt="image" src="https://user-images.githubusercontent.com/13770365/228931889-72b9c8c6-0377-40e3-ac6f-67a0c6a0adea.png">

I'm not sure if this is related to #967. It could be that the visualization isn't working correctly. However, the heatmap of the clusters roughly lines up with the visual, so I think that's less likely. Not sure exactly what's going on. I think a good first step once #967 is resolved will be to re-run on some previous data and confirm that we still get the qualitatively same clustering results, making sure to re-generate the neighbor_counts, rather than using the previously extracted ones.

**Comment by camisowers** [comment]
Looks like the color assignment is accurate based on the outputted kmeans clustering results. Seems like it could be an issue with either the neighbor matrices or distance matrices calculation, which have both been adjusted in the last 4 months. Which previous data should I test out?

**Comment by ngreenwald** [comment]
You could either rerun it on some data, like Erin’s or the example dataset,
where you know what it should like look like. Or you could take the same
data, and run it with the commit from a couple months ago before the
refactoring.

I’m not 100% convinced there’s a problem, but I think there might be. So
just some initial validation to figure out if there’s an obvious issue or
not

**Comment by camisowers** [comment]
I verified that the the generated neighbors matrices have not changed, but it looks like there was an issue with the Kmeans function call itself. 
Scikit-learn 1.2 changed the default `n_init` param from 10 to 'auto', which then caused the algorithm to run only once. On the left is the clustering using a commit from October and the right is using main.

<img width="1043" alt="Screenshot 2023-04-24 at 11 52 42 AM" src="https://user-images.githubusercontent.com/38049893/234089338-6c2b524a-2a6c-45f3-ab00-e0dd42264c1f.png">

I was able to get the same results as before by adding `n_init=10` to the `Kmeans()` call.

<img width="1046" alt="Screenshot 2023-04-24 at 11 43 03 AM" src="https://user-images.githubusercontent.com/38049893/234088760-f7dd9378-66ae-4347-ae16-b4c2e23ddb15.png">

I can open a quick PR now to fix this.
